modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -343,422 +206,15 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
-              text = textNode.value.replace(startWhiteSpaceRegExp, '');
-
-              if (text.length > 0)
-                textNode.value = text;
-              else
-                textNode.remove();
-            }*/
-          }
-
-          // Check if we exited a whitespace preserved element
-          if (isInWhiteSpacePreservedElement && whiteSpaceElements[name]) {
-            isInWhiteSpacePreservedElement = false;
-          }
-
-          if (elementRule.removeEmpty && isEmpty(schema, nonEmptyElements, whiteSpaceElements, node)) {
-            tempNode = node.parent;
-
-            if (blockElements[node.name]) {
-              node.empty().remove();
-            } else {
-              node.unwrap();
-            }
-
-            node = tempNode;
-            return;
-          }
-
-          if (elementRule.paddEmpty && (isPaddedWithNbsp(node) || isEmpty(schema, nonEmptyElements, whiteSpaceElements, node))) {
-            paddEmptyNode(settings, args, blockElements, node);
-          }
-
-          node = node.parent;
-        }
-      }
-    }, schema);
-
-    const rootNode = node = new AstNode(args.context || settings.root_name, 11);
-
-    parser.parse(html, args.format as ParserFormat);
-
-    // Fix invalid children or report invalid children in a contextual parsing
-    if (validate && invalidChildren.length) {
-      if (!args.context) {
-        fixInvalidChildren(invalidChildren);
-      } else {
-        args.invalid = true;
-      }
-    }
-
-    // Wrap nodes in the root into block elements if the root is body
-    if (rootBlockName && (rootNode.name === 'body' || args.isRootContent)) {
-      addRootBlocks();
-    }
-
-    // Run filters only when the contents is valid
-    if (!args.invalid) {
-      // Run node filters
-      for (name in matchedNodes) {
-        if (!Obj.has(matchedNodes, name)) {
-          continue;
-        }
-        list = nodeFilters[name];
-        nodes = matchedNodes[name];
-
-        // Remove already removed children
-        fi = nodes.length;
-        while (fi--) {
-          if (!nodes[fi].parent) {
-            nodes.splice(fi, 1);
-          }
-        }
-
-        for (i = 0, l = list.length; i < l; i++) {
-          list[i](nodes, name, args);
-        }
-      }
-
-      // Run attribute filters
-      for (i = 0, l = attributeFilters.length; i < l; i++) {
-        list = attributeFilters[i];
-
-        if (list.name in matchedAttributes) {
-          nodes = matchedAttributes[list.name];
-
-          // Remove already removed children
-          fi = nodes.length;
-          while (fi--) {
-            if (!nodes[fi].parent) {
-              nodes.splice(fi, 1);
-            }
-          }
-
-          for (fi = 0, fl = list.callbacks.length; fi < fl; fi++) {
-            list.callbacks[fi](nodes, list.name, args);
-          }
-        }
-      }
-    }
+  const parse = (html: string, _args?: ParserArgs): Node => {
+    const parser = new DOMParser();
```

Comment:
This doesn't need to be constructed each time and could be a global instead. That'll help with parse performance.

- [ ] Make this a global variable

Reply:
It's now a (lazily instantiated) global

Reply:
I've left a separate comment about how the lazy instantiation was done so resolving this comment.